---
title: "GWAS_Visual"
output: html_document
date: "2023-10-09"
---

# load library
```{r}
# if not installed run: 
# install.packages("tidyverse")
# install.packages("qqman")
library(tidyverse)
library(qqman)
```


# PCA
```{r}
# Load the sample information for 1000 Genomes (sample ID, population, and super population)
info <- read_tsv("sample_population_info_1kgp.tsv")
#info <- read_tsv("https://raw.githubusercontent.com/OchoaLab/genomic-modules/refs/heads/main/day2/Exercise/data/sample_population_info_1kgp.tsv")
info
```


```{r}
# load PCA results
data <- read_tsv("African.filtered.eigenvec")
#data <- read_tsv("https://raw.githubusercontent.com/OchoaLab/genomic-modules/refs/heads/main/day2/Exercise/result/African.filtered.eigenvec")
data
```


```{r}
# Use the left_join() function to map the sample ID to their corresponding population info:
data <- left_join(data, info, by = c("#IID" = "Sample"))
# better way to inspect big tables:
View(data)
```


```{r}
# label by population info
# PCA plot of the first two components
ggplot(data, aes(x = PC1, y = PC2, color = Population)) + geom_point()
```


# Manhattan Plot

```{r}
# Load the association table:
result <- read_tsv("African.filtered.PHENO1.glm.logistic.hybrid")
# result <- read_tsv("https://raw.githubusercontent.com/OchoaLab/genomic-modules/refs/heads/main/day2/Exercise/result/African.filtered.PHENO1.glm.logistic.hybrid"
# Edit the column names to match what the manhattan function prefers:
colnames(result)[1:3] <- c("CHR","BP","SNP")
head(result)
```


```{r}
# Load the list of causal variants:
causal_variants <- read_lines("variant_ID_APOL1_G1_and_G2.txt")
#causal_variants <- read_lines("https://raw.githubusercontent.com/OchoaLab/genomic-modules/refs/heads/main/day2/Exercise/data/variant_ID_APOL1_G1_and_G2.txt")
causal_variants
```


```{r}
# manhattan plot
manhattan(result, highlight = causal_variants)
```


```{r}
# qqplot
qq(result$P, main = "QQ Plot")
```

